receiver/prometheusreceiver/config.go

```diff
@@ -47,94 +36,74 @@ type Config struct {
-	ConfigPlaceholder any `mapstructure:"config"`
+	TargetAllocator *TargetAllocator `mapstructure:"target_allocator"`
 
 	// EnableProtobufNegotiation allows the collector to set the scraper option for
 	// protobuf negotiation when conferring with a prometheus client.
 	EnableProtobufNegotiation bool `mapstructure:"enable_protobuf_negotiation"`
 }
 
-type targetAllocator struct {
-	Endpoint    string        `mapstructure:"endpoint"`
-	Interval    time.Duration `mapstructure:"interval"`
-	CollectorID string        `mapstructure:"collector_id"`
-	// ConfigPlaceholder is just an entry to make the configuration pass a check
-	// that requires that all keys present in the config actually exist on the
-	// structure, ie.: it will error if an unknown key is present.
-	ConfigPlaceholder any                `mapstructure:"http_sd_config"`
-	HTTPSDConfig      *promHTTP.SDConfig `mapstructure:"-"`
+// Validate checks the receiver configuration is valid.
+func (cfg *Config) Validate() error {
+	if (cfg.PrometheusConfig == nil || len(cfg.PrometheusConfig.ScrapeConfigs) == 0) && cfg.TargetAllocator == nil {
+		return errors.New("no Prometheus scrape_configs or target_allocator set")
+	}
+	return nil
 }
 
-var _ component.Config = (*Config)(nil)
-var _ confmap.Unmarshaler = (*Config)(nil)
-
-func checkFile(fn string) error {
-	// Nothing set, nothing to error on.
-	if fn == "" {
-		return nil
-	}
-	_, err := os.Stat(fn)
-	return err
+type TargetAllocator struct {
+	Endpoint     string            `mapstructure:"endpoint"`
+	Interval     time.Duration     `mapstructure:"interval"`
+	CollectorID  string            `mapstructure:"collector_id"`
+	HTTPSDConfig *PromHTTPSDConfig `mapstructure:"http_sd_config"`
 }
 
-func checkTLSConfig(tlsConfig commonconfig.TLSConfig) error {
-	if err := checkFile(tlsConfig.CertFile); err != nil {
-		return fmt.Errorf("error checking client cert file %q: %w", tlsConfig.CertFile, err)
+func (cfg *TargetAllocator) Validate() error {
+	// ensure valid endpoint
+	if _, err := url.ParseRequestURI(cfg.Endpoint); err != nil {
+		return fmt.Errorf("TargetAllocator endpoint is not valid: %s", cfg.Endpoint)
 	}
-	if err := checkFile(tlsConfig.KeyFile); err != nil {
-		return fmt.Errorf("error checking client key file %q: %w", tlsConfig.KeyFile, err)
+	// ensure valid collectorID without variables
+	if cfg.CollectorID == "" || strings.Contains(cfg.CollectorID, "${") {
+		return fmt.Errorf("CollectorID is not a valid ID")
 	}
+
 	return nil
 }
 
-// Validate checks the receiver configuration is valid.
-func (cfg *Config) Validate() error {
-	promConfig := cfg.PrometheusConfig
-	if (promConfig == nil || len(promConfig.ScrapeConfigs) == 0) && cfg.TargetAllocator == nil {
-		return errors.New("no Prometheus scrape_configs or target_allocator set")
-	}
+// PromConfig is a redeclaration of the promconfig.Config because we need custom unmarshaling
+// because prometheus "config" uses `yaml` tags.
```

Comment:
[Nit] Sligthly improved language IMO.
```suggestion
// PromConfig is a redeclaration of promconfig.Config because we need custom unmarshaling
// as prometheus "config" uses `yaml` tags.
```

Reply:
Done.